torch/_export/passes/lift_constants_pass.py

```diff
@@ -151,7 +151,7 @@ def lift_constants_pass(
                 constant_kind = InputKind.CUSTOM_OBJ
                 constant_fqn = constant_attrs.get(constant_val)
                 if constant_fqn is not None:
-                    _, _, constant_name = constant_fqn.rpartition(".")
+                    constant_name = constant_fqn.replace(".", "_")
```

Comment:
@zzh142857 If we have two constant fqns like the following:
```
self.nested_attr.constant
```
and
```
self.nested.attr_constant
```
the name collision is still there. I think there should be a better way to handle collision.

I saw the only use of `constant_name` is `graph.placeholder(constant_name)`. I'm wondering what's the exact error you saw in the first place?

Reply:
Oh good point. The error I got is when we have multiple instances of the same class who contains constant tensors, something like: `self.tbe_1.weights_dev` and `self.tbe_2.weights_dev`. When they are lifted as input args, the previous logic will create two placeholders with the same target name `weights_dev` therefore the graph cannot be compiled.

Reply:
@zhxchen17 Probably I can still keep the previous logic and add a global counter here to track, then just add a suffix for duplicated constant_name?

Reply:
could you send me the error message you saw in the first place?

> Probably I can still keep the previous logic and add a global counter here to track, then just add a suffix for duplicated constant_name?

That's fine, but I remember fx already does this for you, so I'm confused why you need to do this at user level.

Reply:
Yeah seems fx can add suffix for node name, but not the target/arg of the node, so for my case I will get two placeholders like this: 
```
%weights_dev : [num_users=1] = placeholder[target=weights_dev]
%weights_dev_1 : [num_users=1] = placeholder[target=weights_dev]
``` 
they do have different node.name, but still have collision on their targets.

Reply:
oh ok I get what you mean now. In this case you set reset the node target like
```
node.target = node.name # assuming node.op == 'placeholder'
```

Ideally we should fix this in FX by adding a new argument to .placeholder call.

Reply:
Oh thanks, yeah this is much cleaner solution! let me update